junit/mockwebserver/src/main/java/io/fabric8/mockwebserver/internal/MockServerExpectationImpl.java

```diff
@@ -224,6 +234,21 @@ private void enqueue(ServerRequest req, ServerResponse resp) {
   private ServerResponse createResponse(boolean repeatable, long delay, TimeUnit delayUnit) {
     if (chunksProvider != null) {
       return new ChunkedResponse(repeatable, delay, delayUnit, chunksProvider);
+    } else if (failing) {
+      return new ServerResponse() {
+        @Override
+        public boolean isRepeatable() {
+          return repeatable;
+        }
+
+        @Override
+        public MockResponse toMockResponse(RecordedRequest recordedRequest) {
+          MockResponse mockResponse = new MockResponse();
+          mockResponse.setResponseCode(500);
+          mockResponse.setSocketPolicy(SocketPolicy.DISCONNECT_AFTER_REQUEST);
+          return mockResponse;
+        }
```

Comment:
This is quite blunt but and feels limited (there are many other possible socket policies) but I don't see a clean way of exposing them through the DSL and I don't have a specific use case for any others at the moment.

Reply:
I've killed off the changes to the mockwebserver and replaced them with a dead socket so the initial connection always fails.